src/components/Calculator.js

```diff
@@ -1,139 +1,149 @@
             className="items item7 light-bg"
-            onClick={() => handleClick('9')}
+            onClick={(e) => clickHandler(e.target.textContent)}
           >
             9
           </button>
           <button
             type="button"
             className="items item8 golden-bg"
-            onClick={() => handleClick('x')}
+            onClick={(e) => clickHandler(e.target.textContent)}
           >
             x
           </button>
           <button
             type="button"
             className="items item9 light-bg"
-            onClick={() => handleClick('4')}
+            onClick={(e) => clickHandler(e.target.textContent)}
           >
             4
           </button>
           <button
             type="button"
             className="items item10 light-bg"
-            onClick={() => handleClick('5')}
+            onClick={(e) => clickHandler(e.target.textContent)}
           >
             5
           </button>
           <button
             type="button"
             className="items item11 light-bg"
-            onClick={() => handleClick('6')}
+            onClick={(e) => clickHandler(e.target.textContent)}
           >
             6
           </button>
           <button
             type="button"
             className="items item12 golden-bg"
-            onClick={() => handleClick('+')}
+            onClick={(e) => clickHandler(e.target.textContent)}
           >
             +
           </button>
           <button
             type="button"
             className="items item13 light-bg"
-            onClick={() => handleClick('1')}
+            onClick={(e) => clickHandler(e.target.textContent)}
           >
             1
           </button>
           <button
             type="button"
             className="items item14 light-bg"
-            onClick={() => handleClick('2')}
+            onClick={(e) => clickHandler(e.target.textContent)}
           >
             2
           </button>
           <button
             type="button"
             className="items item15 light-bg"
-            onClick={() => handleClick('3')}
+            onClick={(e) => clickHandler(e.target.textContent)}
           >
             3
           </button>
           <button
             type="button"
             className="items item16 golden-bg"
-            onClick={() => handleClick('-')}
+            onClick={(e) => clickHandler(e.target.textContent)}
           >
             -
           </button>
           <button
             type="button"
             className="items item17 light-bg"
-            onClick={() => handleClick('0')}
+            onClick={(e) => clickHandler(e.target.textContent)}
           >
             0
           </button>
           <button
             type="button"
             className="items item18 golden-bg"
-            onClick={() => handleClick('=')}
+            onClick={(e) => clickHandler(e.target.textContent)}
           >
             =
           </button>
```

Comment:
[OPTIONAL]
To optimize the given code, we can reduce the redundancy by mapping over an array of button values and rendering them programmatically. Here's an optimized version of the `Calculator` component:

```jsx
import React, { useState } from 'react';
import calculate from '../logic/calculate';
import Result from './Result';

const Calculator = () => {
  const [value, setValue] = useState({
    total: '0',
    next: null,
    operation: null,
  });
  const { total, operation, next } = value;

  function clickHandler(num) {
    setValue(calculate(value, num));
  }

  const buttons = [
    ['AC', 'light-bg'], ['+/-', 'light-bg'], ['%', 'light-bg'], ['/', 'golden-bg'],
    ['7', 'light-bg'], ['8', 'light-bg'], ['9', 'light-bg'], ['x', 'golden-bg'],
    ['4', 'light-bg'], ['5', 'light-bg'], ['6', 'light-bg'], ['+', 'golden-bg'],
    ['1', 'light-bg'], ['2', 'light-bg'], ['3', 'light-bg'], ['-', 'golden-bg'],
    ['0', 'light-bg'], ['=', 'golden-bg']
  ];

  return (
    <div className="math-container">
      <div className="container">
        <h2 className="heading">Lets do some Math</h2>
        <section className="section">
          <div className="results gray-bg flex-container">
            <Result result={next || operation || total || '0'} />
          </div>
          {buttons.map((btn, index) => (
            <button
              key={index}
              type="button"
              className={`items item${index + 1} ${btn[1]}`}
              onClick={() => clickHandler(btn[0])}
            >
              {btn[0]}
            </button>
          ))}
        </section>
      </div>
    </div>
  );
};

export default Calculator;
```

By using the `buttons` array and the `map` function, we reduce the repetitive nature of the button creation and make the code much more concise and readable.

Reply:
Thank you for the recommendation